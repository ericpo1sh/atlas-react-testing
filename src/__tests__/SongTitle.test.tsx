import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import SongTitle from "../components/SongTitle";
import { MusicContext } from "../Context/MusicContext";

test("Song Title Renders Correctly On Fail", () => {
  const { container } = render(<SongTitle/>);
  expect(container).toMatchSnapshot();
});

const mockContextValue = {
  CurrentSong: {
    cover: "https://m.media-amazon.com/images/I/71pxGj4RoVS._AC_UF894,1000_QL80_.jpg",
    title: "Graduation",
  },
};

test("Song Title Renders Correctly On Success", () => {
  const { container } = render(
    <MusicContext.Provider value={mockContextValue}>
      <SongTitle />
    </MusicContext.Provider>
  );

  const titleElement = screen.getByText(/Graduation/i);
  expect(titleElement).not.toBeNull();
  expect(container).toMatchSnapshot();
})
