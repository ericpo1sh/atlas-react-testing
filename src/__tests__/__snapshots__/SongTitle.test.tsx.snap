// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Song Title Renders Correctly On Fail 1`] = `
<div>
  <div>
    Not Found
  </div>
</div>
`;

exports[`Song Title Renders Correctly On Success 1`] = `
<div>
  <div
    class="flex-col flex items-center mt-4"
  >
    <div
      class="w-96 text-left"
    >
      <h1
        class="text-2xl font-bold"
      >
        Graduation
      </h1>
      <p
        class="text-jacarta-500 font-medium"
      />
    </div>
  </div>
</div>
`;
